src/string.ts

```diff
@@ -16,3 +16,80 @@ export function parseBoolean(candidate: string | number) {
     return false;
   }
 }
+
+/**
+ * Splits a string into an array of words, respecting quoted phrases.
+ * Handles comma-separated values where phrases in double quotes are treated as a single unit,
+ * even if they contain commas.
+ *
+ * This is basically CSV parsing, if we find this to be cumbersome to maintain,
+ * we can consider using a dedicated CSV parsing library.
+ *
+ * Example: '"apple, banana", cherry' => ['apple, banana', 'cherry']
+ *
+ * @param {string} candidate - The input string to split
+ * @return {string[]} - Array of split words/phrases
+ */
+export function splitWords(candidate: string): string[] {
+  // Handle empty input case
+  if (!candidate) return [''];
```

Comment:
Should we return an empty array here or array with empty string?

Reply:
I _was_ not sure exactly what to do. I assumed that it will do an empty check like `[installation_type='']` but I guess `nil` would work too. 

I'll change this behavior